src/util/SE3.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+from tf_utils import vector3_to_point, point_to_vector3
+from ros_numpy import numpify
+from gettext import translation
+import numpy as np
+from geometry_msgs.msg import Pose, Transform
+from tf.transformations import quaternion_matrix, quaternion_inverse, \
+    quaternion_multiply, rotation_from_matrix
+
+
+class SE3(Pose):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @classmethod
+    def from_tf(cls, tf: Transform) -> SE3:
+        """
+        Create an SE3 pose object from a Transform message.
+        The translation of the Transform will become the position of the Pose,
+        and the rotation of the Transform will become the orientation of the Pose.
+
+        Args:
+            tf: the Transform message object
+
+        Returns:
```

Comment:
Do you think you can use the pydoc official formatting?

Like :return: and stuff: https://stackoverflow.com/questions/34331088/how-to-comment-parameters-for-pydoc

Reply:
done